src/node.rb

```diff
@@ -68,8 +73,14 @@ def built?
     def groups
       NodeattrInterface.groups_for_node(name)
     rescue NodeNotInGendersError
-      # It's OK for a node to not be in the genders file, it just means it's
-      # not part of any groups.
+      # Re-raise if we're expecting the node to be configured, as it should be
+      # in the genders file.
+      raise if should_be_configured
+
+      # The node doesn't need to be in the genders file if it's not expected to
+      # be configured, it's just part of no groups yet (XXX not sure if this is
+      # still true, maybe we should be stricter now and always require a node
+      # to be in the genders file).
       []
```

Comment:
I think you may be right, I've added a card to track this here: https://trello.com/c/mMvRUUxw/149-always-require-created-nodes-exist-in-the-genders-file